### PR TITLE
Harden partest against duplicate paths.

### PR DESCRIPTION
**Pull request by paulp**
A serious issue: partest would launch the same test multiple times if a test
path was given in multiple forms (e.g. absolute and relative paths.)
Unfortunately all those tests would share the same logfile, output directory,
etc. which would predictably lead to explosions.

Since overwriting classfiles while being loaded can lead to jvm core dumps,
it's possible this is involved in recent jvm crashes and other test
breakdowns.

This commit also alters the default partest verbosity to only full print test
transcripts under --verbose.

### CODE REVIEW

**Comment by paulp** [comment]
This is a more tightly focused variation on #2567 because I suddenly realize we better get this in pronto.

**Comment by paulp** [comment]
Review by @xeno-by

**Comment by xeno-by** [comment]
LGTM

File: src/partest/scala/tools/partest/nest/ConsoleRunner.scala [hunk 1]
```diff
@@ -185,8 +192,8 @@ class ConsoleRunner extends DirectRunner {
 
     chatty(banner)
 
-    val allTests   = (miscTests ++ (kinds flatMap testsFor)).distinct
-    val grouped    = (allTests groupBy kindOf).toList sortBy (x => standardKinds indexOf x._1)
+    val allTests: List[Path] = distinctBy(miscTests ++ kindsTests)(_.toCanonical) sortBy (_.toString)
```
Comment (xeno-by) [inline src/partest/scala/tools/partest/nest/ConsoleRunner.scala:195]: How come we got same tests multiple times here?

Comment (paulp) [inline src/partest/scala/tools/partest/nest/ConsoleRunner.scala:195]: Well, one way is for a test to have failed (so it's in rerunTests) but also to be specified or a member of a given category.

Comment (paulp) [inline src/partest/scala/tools/partest/nest/ConsoleRunner.scala:195]: Because rerunTests, undesirably, have their test paths absolute.

**Comment by paulp** [comment]
Fasttracking.

**Pull request merged**